src/validate.ts

```diff
@@ -155,17 +172,36 @@ export function validate(
           isBinaryArithmeticToken(node.operatorToken.kind)) ||
           ((ts.isPrefixUnaryExpression(node) ||
             ts.isPostfixUnaryExpression(node)) &&
-            (isUnaryArithmeticToken(node.operator) ||
-              // we can support unary `+` because it does not change the value, but unary `-` does.
-              node.operator === ts.SyntaxKind.MinusToken))) &&
+            isUnaryArithmeticToken(node.operator))) &&
         !checker.isConstant(node)
       ) {
-        return [
-          newError(
-            node,
-            ErrorCodes.Cannot_perform_arithmetic_or_bitwise_computations_on_variables_in_Step_Function
-          ),
-        ];
+        if (
+          (ts.isBinaryExpression(node) &&
+            isBinaryArithmeticToken(node.operatorToken.kind) &&
+            StepFunctions.SupportedBinaryArithmeticToken.has(
+              node.operatorToken.kind
+            )) ||
+          ((ts.isPrefixUnaryExpression(node) ||
+            ts.isPostfixUnaryExpression(node)) &&
+            isUnaryArithmeticToken(node.operator) &&
+            StepFunctions.SupportedUnaryBinaryArithmeticToken.has(
+              node.operator
+            ))
+        ) {
+          return [
+            newWarning(
+              node,
+              ErrorCodes.Step_Functions_Arithmetic_Only_Supports_Integers
+            ),
+          ];
+        } else {
+          return [
+            newError(
+              node,
+              ErrorCodes.Cannot_perform_arithmetic_or_bitwise_computations_on_variables_in_Step_Function
```

Comment:
Did we update this error code with the new addition change?

Reply:
the logic, forgot about the text.